web/ui/react-app/src/ExpressionInput.tsx

```diff
@@ -67,54 +71,62 @@ class ExpressionInput extends Component<ExpressionInputProps, ExpressionInputSta
     }
   }
 
-  executeQuery = () => this.props.executeQuery(this.exprInputRef.current!.value)
+  executeQuery = () => this.props.executeQuery(this.exprInputRef.current!.value);
 
-  renderAutosuggest = (downshift: ControllerStateAndHelpers<any>) => {
-    const { inputValue } = downshift
+  getSearchMatches = (input: string, expressions: string[]) => {
+    return fuzzy.filter(input.replace(/ /g, ''), expressions, {
+      pre: "<strong>",
+      post: "</strong>",
+    });
+  }
+
+  creatAutocompleteSection = (downshift: ControllerStateAndHelpers<any>) => {
+    const { inputValue, closeMenu, highlightedIndex } = downshift;
     if (!inputValue || (this.prevNoMatchValue && inputValue.includes(this.prevNoMatchValue))) {
```

Comment:
@juliusv what is the purpose of ```(this.prevNoMatchValue && inputValue.includes(this.prevNoMatchValue))```
Seems like everything is fine when I remove it

Reply:
I added this hack back then to skip the fuzzy-matching calculation, which can create noticeable delays while typing when there are a lot of metrics to match against.

When we already know that the previous input didn't have a match entry and the new input value is a superset of the old one, it will definitely also not match anything.